pinot-common/src/main/java/org/apache/pinot/common/segment/generation/SegmentGenerationUtils.java

```diff
@@ -229,4 +233,63 @@ private static String fetchUrl(URL url, String authToken)
     }
     return IOUtils.toString(connection.getInputStream(), StandardCharsets.UTF_8);
   }
+
+
+  /**
+   * Find matching files from root directory specified in fileUri.
+   * If includePattern and excludePattern are not null, get all the files that match includePattern and exclude files
+   * that match excludePattern.
+   * If
+   *
+   * @param pinotFs root directly fs
+   * @param fileUri root directly uri
+   * @param includePattern optional glob patterns for files to include
+   * @param excludePattern optional glob patterns for files to exclude
+   * @param searchRecrusively if ture, search files recursively from directory specified in fileUri
+   * @return list of matching files.
+   * @throws IOException on IO failure for list files in root directory.
+   * @throws URISyntaxException for matching file URIs
+   * @throws RuntimeException if there is no matching file.
+   */
+  public static List<String> listMatchedFilesWithRecursiveOption(PinotFS pinotFs, URI fileUri, String includePattern,
```

Comment:
Annotate `includePattern` and `excludePattern` as `Nullable`

Reply:
Done